### PR TITLE
Search movies functionality implemented

### PR DESCRIPTION
**Pull request by HoracioDuca**
### TRENDING MOVIES API

**New features implemented**:

- _Search_ function: Now you can search for a movie or actor that you want.
- The search function results are provided by fetching the TMDB API.
- UI for filter fetched movies was created.
- Search button now has a transition to show title or search text.
- When you press the "X" button on search you clean the query.

**SEARCH FUNCTIONALITY**

movies

**SEARCH BUTTON, TITLE TRANSITION AND CANCEL BUTTON CLEANS QUERY**

movies2

### CODE REVIEW

File: lib/src/blocs/i_movies_bloc.dart [hunk 1]
```diff
@@ -5,4 +5,8 @@ abstract class IMoviesBloc extends IBloc {
   Stream<Movie> get streamMovies;
 
   void fetchMoviesApi();
+
+  void fetchMoviesFilter(
+    String filterKeys,
```
Comment (FranciscoGaraventa) [inline lib/src/blocs/i_movies_bloc.dart]: You can change this parameter name to query instead of filterKeys

File: lib/src/blocs/movies_bloc.dart [hunk 2]
```diff
@@ -19,9 +19,26 @@ class MoviesBloc extends IMoviesBloc {
   @override
   void fetchMoviesApi() async {
```
Comment (FranciscoGaraventa) [inline lib/src/blocs/movies_bloc.dart]: You can change this function with a more descriptive name, remember that don't reference to Api. You can use the same approach of _movieRepository fetch call (fetchAllMovies)

File: lib/src/models/result_movie.dart [hunk 3]
```diff
@@ -16,12 +18,12 @@ class ResultMovie {
 
   ResultMovie({
     this.voteCount,
-    required this.id,
+    this.id,
     this.video,
     this.voteAverage,
     this.title,
     this.popularity,
-    required this.posterPath,
```
Comment (FranciscoGaraventa) [inline lib/src/models/result_movie.dart]: Why did you remove required from id and posterPath? Why the other parameters are not required?

File: lib/src/blocs/i_movies_bloc.dart (hunk 1 above)
Comment (HoracioDuca) [reply lib/src/blocs/i_movies_bloc.dart]: done

File: lib/src/blocs/movies_bloc.dart (hunk 2 above)
Comment (HoracioDuca) [reply lib/src/blocs/movies_bloc.dart]: Thank you for remembering me that. More descriptive name added.

**Review by FranciscoGaraventa — COMMENTED** [review]
Try to add a gif that show the logic that u used when cancel button is pressed, to see how you show Title text or not

File: lib/src/models/result_movie.dart (hunk 3 above)
Comment (HoracioDuca) [reply lib/src/models/result_movie.dart]: Because I had a problem when I wanted to show a default image when posterPath response was null, but in the end it had nothing to do with it. So I already changed all the parameters  -those who can-  to be required.

**Comment by HoracioDuca** [comment]
> 
> 
> Try to add a gif that show the logic that u used when cancel button is pressed, to see how you show Title text or not

Added in description.

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by FranciscoGaraventa — CHANGES_REQUESTED** [review]
With your implementation you lose the trending movies call,when you press on cancel button you should clean the query so it's going to show again trending movies on home page

**Review by marianosalvetti — APPROVED** [review]

**Comment by HoracioDuca** [comment]
> 
> 
> With your implementation you lose the trending movies call,when you press on cancel button you should clean the query so it's going to show again trending movies on home page

Done.

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM
